### PR TITLE
World modification fix

### PR DESCRIPTION
**Pull request by Earthcomputer**
`PlayerEntity.canMine` was inverted.
"Restricted" comes from a string in `ServerPlayerInteractionManager.processBlockBreakingAction`.

### CODE REVIEW

**Comment by YanisBft** [comment]
Hum, this method name looks like it belongs to a World class. What about `isPreventedFromWorldModification`?

**Comment by kvverti** [comment]
`canMine` -> `cannotMine`
`shouldLimitWorldModification` -> `shouldAllowWorldModification`

**Comment by Earthcomputer** [comment]
> this method name looks like it belongs to a World class

Does it really? Looks obvious to me.

> `canMine` -> `cannotMine`

I don't like putting negatives into names, it gets confusing as soon as a `!` introduces a double negative. "Restricted" is just as clear as "cannot mine", and "not restricted" is 100x clearer than "not cannot mine".

> `shouldLimitWorldModification` -> `shouldAllowWorldModification`

No, that would be an incorrect name.

**Comment by kvverti** [comment]
Ah I thought both had changed, since you renamed both methods. In that case `shouldLimitWorldModification` should remain unchanged because it is out of scope.

**Comment by liach** [comment]
`isBlockBreakingRestricted`? `hasRestrictedInteraction`?

**Comment by Earthcomputer** [comment]
People seem to be randomly suggesting alternative names without explaining why they are better than mine

**Comment by liach** [comment]
This is only affecting block breaking. `world modification` is too broad for this, like I'd argue killing mobs is world modification as well.

**Comment by Runemoro** [comment]
@liach Just block breaking, or placing too?

Also, this method doesn't seem to take a `player` parameter. If the player is in adventure mode and they're holding a tool with an NBT tag that allows breaking certain blocks, this method would return true, but breaking blocks is still allowed.

**Comment by liach** [comment]
block breaking only, used in player interaction manager or server play network handler

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/entity/player/PlayerEntity.mapping [hunk 1]
```diff
@@ -134,7 +134,7 @@ CLASS net/minecraft/class_1657 net/minecraft/entity/player/PlayerEntity
 		ARG 2 inventory
 	METHOD method_7292 getLuck ()F
 	METHOD method_7293 vanishCursedItems ()V
-	METHOD method_7294 canModifyWorld ()Z
+	METHOD method_7294 canModifyBlocks ()Z
```
Comment (liach) [inline mappings/net/minecraft/entity/player/PlayerEntity.mapping:137]:
```suggestion
	METHOD method_7294 canBreakBlocks ()Z
```
unless this is used for interaction checks like axe right click/shovel right click

Comment (Earthcomputer) [reply mappings/net/minecraft/entity/player/PlayerEntity.mapping:137]: It's used for interaction checks where the interaction may modify the block, like inverting a daylight sensor, as well as for block breaking

**Review by liach — APPROVED** [review]

**Review by hYdos — APPROVED** [review]
looks good. naming looks fine

**Review by Juuxel — APPROVED** [review]

**Pull request merged**